docs/design/auth.md

```diff
@@ -0,0 +1,52 @@
+# Runner Authentication and Authorization
```

Comment:
add sequence diagram for config/run/hosted

Reply:
https://www.websequencediagrams.com/

Reply:
Incoming! 😄

Reply:
Also, checkin the raw text from websequencediagrams into a resource folder.

Reply:
Done! 👍